### ISSUE TITLE
Should we specify attention_mask when using gpt2 for the conversation task?

### ISSUE THREAD

**Issue by Oran-Ac** [RUCAIBox/CRSLab#41]
Here's an example when using `'redial dataset'`
We do the padding function to make the dialogue in the same length, so that we can deal with batches.

- Dataloader process: [code](https://github.com/RUCAIBox/CRSLab/blob/main/crslab/data/dataloader/tgredial.py#L292~#L300)

In order to make the gpt2 pay no attention to the `pad`, should we specify attention_mask when using gpt2 for conversation task?
- gpt2 forward process:[code](https://github.com/RUCAIBox/CRSLab/blob/main/crslab/model/conversation/gpt2/gpt2.py#L71)
- helpful issues: [link](https://github.com/huggingface/transformers/issues/2630#issuecomment-684512764)

Because we calculate the loss just with the [response](https://github.com/RUCAIBox/CRSLab/blob/main/crslab/model/conversation/gpt2/gpt2.py#L74~#L76), should the padding labels set  to `-100` rather than `0`([code](https://github.com/RUCAIBox/CRSLab/blob/main/crslab/data/dataset/redial/resources.py#L56)) so that the model can ignore it?

**Comment by Oran-Ac** [comment]
I think when doing the batch generation, we also need to pass the `position ids` to the model. Otherwise, the `position ids` will always be `1`  in the loop except for the first round. You can find the discussion [here].(https://github.com/huggingface/transformers/issues/3021#issuecomment-591418233)

https://github.com/RUCAIBox/CRSLab/blob/b3ab262a4ad93cbae98fe66541eb735377768a35/crslab/model/conversation/gpt2/gpt2.py#L98

And why you cut off the context when doing the generation?
https://github.com/RUCAIBox/CRSLab/blob/b3ab262a4ad93cbae98fe66541eb735377768a35/crslab/model/conversation/gpt2/gpt2.py#L95

**Comment by txy77** [comment]
Thanks for your feedback. @Oran-Ac 
1. The padding labels are set to **-100** according to your suggestion.
2. Due to the limitation of gpu memory, we cuf off the context when doing the generation.
3. The position_ids will be automatically added by Transformers if no parameter is passed.
